web_src/js/bootstrap.js

```diff
@@ -40,7 +48,7 @@ function initGlobalErrorHandler() {
     processWindowErrorEvent(e);
   }
   // then, change _globalHandlerErrors to an object with push method, to process further error events directly
-  window._globalHandlerErrors = {'push': (e) => processWindowErrorEvent(e)};
+  window._globalHandlerErrors = {_inited: true, push: (e) => processWindowErrorEvent(e)};
 }
```

Comment:
@wxiaoguang could you port the Proxy implementation from https://github.com/go-gitea/gitea/pull/25940 here instead of extending this hack further? I think https://github.com/go-gitea/gitea/pull/25940 is not going to land anytime soon, so it's a good extract.

Reply:
Hmm, I have different opinion. I think this "array" is a right approach. The reasons are:

1. The bug you met is not caused by the array, but by duplicate-init. I have added "inited" check in this PR, so you won't see any new problem.
2. Array is also widely used by big company products, for example, Google Analytics also uses `ga.push()`, the same method.
3. Using an object with "push" method is light than using a proxy object, and it doesn't need too much tricky code in the "set" method.

So, I would always prefer this approach.

Reply:
Does your hack even run in JS strict mode? Try adding `"use strict";` to the file 😆.

Reply:
It's not hack IMO, it strictly follows JS standard. And, why not not? I don't see any problem.

```

<script>
	window._testGlobal = [];
	window._testGlobal.push('foo');
</script>
<script src="{{AssetUrlPrefix}}/test.js"></script>
```

```js
"use strict";

function test() {
  for (const e of window._testGlobal || []) {
    console.log(`test (pre): ${e}`);
  }
  window._testGlobal = {_inited: true, push: (e) => console.log(`test (after): ${e}`)};
}
test();

window._testGlobal.push('bar');
```
![image](https://github.com/go-gitea/gitea/assets/2114189/97b79749-9553-4a14-9cbb-d634f41cf9de)

Reply:
I guess you are right and the problem I've been seeing may be because of that missing init. Still I personally prefer the Proxy method.

BTW, I would like our code to run in strict mode, but don't feel like littering every file with a "use strict". Maybe there are solutions for webpack to automatically inject it into every chunk.

Reply:
According to https://github.com/evanw/esbuild/issues/422#issuecomment-739740602 we could use the esbuild banner feature for it.